### PR TITLE
Manage artifacts used by us and other Quarkus Platform participants

### LINKED ISSUE #26068: Manage com.squareup.okio:okio
### Describe the bug

We manage `com.squareup.okhttp3:okhttp` but we do not manage its major dependency `com.squareup.okio:okio` of which other platform participants may bring an incompatible version.

Hence we should start managing `com.squareup.okio:okio`.

A fix follows.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### LINKED ISSUE #26071: Ban org.javassist:javassist
### Describe the bug

Similar case as #26066

quarkus-scheduler depends on org.javassist:javassist but we do not manage it and so other platform participants may bring an incompatible version.

Hence we should start managing org.javassist:javassist.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### LINKED ISSUE #26072: Manage com.google.api.grpc:proto-google-common-protos
### Describe the bug

Similar case as #26066 and #26071

Required by quarkus-grpc (supported) but proto-google-common-protos is not managed in io.quarkus:quarkus-bom and so other platform participants may bring an incompatible version.

Hence we should start managing com.google.api.grpc:proto-google-common-protos

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### LINKED ISSUE #26073: Manage org.hdrhistogram:HdrHistogram
### Describe the bug

Similar case as #26066 #26071 and #26072

Required by quarkus-micrometer (supported), but not managed by Quarkus core and so other platform participants may bring an incompatible version.

Hence we should start managing org.hdrhistogram:HdrHistogram

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### LINKED ISSUE #26087: Manage com.sun.mail:jakarta.mail
### Describe the bug

Similar case as https://github.com/quarkusio/quarkus/pull/26066 https://github.com/quarkusio/quarkus/issues/26071 https://github.com/quarkusio/quarkus/issues/26072 and https://github.com/quarkusio/quarkus/issues/26073

com.sun.mail:jakarta.mail is required by quarkus-keycloak-admin-client and quarkus-resteasy-multipart, but not managed by Quarkus core and so other platform participants may bring an incompatible version.

Hence we should start managing com.sun.mail:jakarta.mail

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #26068
Fix #26071
Fix #26072
Fix #26073
Fix #26087

### CODE REVIEW

**Comment by ppalaga** [comment]
See also https://issues.redhat.com/browse/QUARKUS-2213

A backport to 2.7 would be nice

File: bom/application/pom.xml [hunk 1]
```diff
@@ -189,6 +191,7 @@
         <apicurio-registry.version>2.2.4.Final</apicurio-registry.version>
         <apicurio-common-rest-client.version>0.1.9.Final</apicurio-common-rest-client.version> <!-- must be the version Apicurio Registry uses -->
         <jacoco.version>0.8.8</jacoco.version>
+        <javassist.version>3.27.0-GA</javassist.version>
```
Comment (gastaldi) [inline bom/application/pom.xml:194]: Where is this property being used?

Comment (ppalaga) [reply bom/application/pom.xml:194]: Ups, thanks for catching this. Apparently the change in the BOM is missing.

File: bom/application/pom.xml [hunk 2]
```diff
@@ -61,6 +62,7 @@
         <jakarta.interceptor-api.version>1.2.5</jakarta.interceptor-api.version>
         <jakarta.json.version>1.1.6</jakarta.json.version>
         <jakarta.json.bind-api.version>1.0.2</jakarta.json.bind-api.version>
+        <jakarta.mail.version>1.6.7</jakarta.mail.version>
```
Comment (gastaldi) [inline bom/application/pom.xml:65]: Where is this property being used?

**Comment by geoand** [comment]
The title of the PR looks misleading based on the number of changes of changes it introduces

**Comment by ppalaga** [comment]
> The title of the PR looks misleading based on the number of changes of changes it introduces
Renamed

**Review by gsmet — APPROVED** [review]
Looks good overall but let's get another CI run to be on the safe side.

**Comment by ppalaga** [comment]
@gsmet Do I need to investigate the failing tests?

File: bom/application/pom.xml (hunk 2 above)
Comment (gsmet) [reply bom/application/pom.xml:65]: Yeah this one looks weird.

**Review by gsmet — CHANGES_REQUESTED** [review]
@ppalaga could you have a look at @gastaldi 's question? Let's see what CI says once it's rebased.
The failures look odd but it might be an infrastructure flake.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!

**Comment by gsmet** [comment]
@ppalaga do you think this is still needed for the next 2.7?
@harshmadhani could you check if it's going to be a problem for prod? It might not be as it probably just points to the version we are currently using but let's make sure it doesn't cause you headaches.

**Comment by harshmadhani** [comment]
@gsmet Yes, since there is no change in versions, it should okay for prod. Thanks !

**Comment by ppalaga** [comment]
> @ppalaga do you think this is still needed for the next 2.7?

No, in 2.7 we have some workarounds in place that are fine to live with till its EOL

**Pull request merged**